sql-plugin/src/main/scala/org/apache/spark/sql/rapids/HashFunctions.scala

```diff
@@ -74,6 +74,101 @@ case class GpuSha1(child: Expression)
+  extends GpuBinaryExpression with ImplicitCastInputTypes with NullIntolerantShim {
+  private val childExpr: Expression = left
+  private val bitLengthExpr: Expression = right
+  override def toString: String = s"sha2($childExpr, $bitLength)"
+  override def inputTypes: Seq[AbstractDataType] = Seq(BinaryType, IntegerType)
+  override def dataType: DataType = StringType
+
+  private val bitLength: Option[Int] = bitLengthExpr match {
+    case lit: GpuLiteral if lit.value.isInstanceOf[Int] =>
+      lit.value.asInstanceOf[Int] match {
+        case 224 | 256 | 384 | 512 => Some(lit.value.asInstanceOf[Int])
+        case 0 => Some(256) // Spark default
+        case _ => None // SHA2 returns null for other bit lengths.
+      }
+    case _ => None
+  }
+
+  override def nullable: Boolean = childExpr.nullable || bitLength.isEmpty
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
+    if (bitLength.isEmpty) {
+      // SHA2 returns null for "unsupported" bit lengths.
+      withResource(Scalar.fromNull(DType.STRING)) { nullStringExemplar =>
+        ColumnVector.fromScalar(nullStringExemplar, lhs.getRowCount.toInt)
+      }
+    } else {
+        bitLength match {
+          case Some(224) => withResource(GpuSha2.getStringViewOfBinaryColumn(lhs.getBase)) {
+            Hash.sha224NullsPreserved(_)
+          }
+          case Some(256) => withResource(GpuSha2.getStringViewOfBinaryColumn(lhs.getBase)) {
+            Hash.sha256NullsPreserved(_)
+          }
+          case Some(384) => withResource(GpuSha2.getStringViewOfBinaryColumn(lhs.getBase)) {
+            Hash.sha384NullsPreserved(_)
+          }
+          case Some(512) => withResource(GpuSha2.getStringViewOfBinaryColumn(lhs.getBase)) {
+            Hash.sha512NullsPreserved(_)
+          }
+          case unexpected =>
+            throw new UnsupportedOperationException(s"Unsupported bit length for SHA2: $unexpected")
+        }
+    }
+  }
+
+  override def doColumnar(numRows: Int, lhs: GpuScalar, rhs: GpuScalar): ColumnVector = {
+    withResource(GpuColumnVector.from(lhs, numRows, lhs.dataType)) {
+      doColumnar(_, rhs)
+    }
+  }
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    throw new UnsupportedOperationException("Only literal bit lengths are supported for SHA2")
+  }
+
+  override def doColumnar(lhs: GpuScalar, rhs: GpuColumnVector): ColumnVector = {
+    throw new UnsupportedOperationException("Only literal bit lengths are supported for SHA2")
+  }
+}
+
+object GpuSha2 {
+
+  private def getStringViewOfBinaryColumn(col: ColumnView): ColumnVector = {
+    withResource(col.getChildColumnView(0)) { dataCol =>
+      withResource(new ColumnView(DType.STRING, col.getRowCount,
+        Optional.of[java.lang.Long](col.getNullCount),
+        dataCol.getData, col.getValid, col.getOffsets)) { cv =>
+        cv.copyToColumnVector()
+      }
+    }
+  }
+
+  def getMeta(e: Sha2,
```

Comment:
NIT:
Can we create a named class instead of the method returning a anonymous class ? e.g.
```
class Sha2Meta(e, conf, p, r) extends BinaryExprMeta[Sha2](e, conf, p, r) {
  override def convertToGpu(lhs: Expression, rhs: Expression): GpuExpression = GpuSha2(lhs, rhs)
}
```
Then in the GpuOverrides, it becomes
```
(a, conf, p, r) => new Sha2Meta(a, conf, p, r)
```

Reply:
I've reworked the getter function into a named class.  This is more descriptive.